[DataGridPro] Allow non-pinnable columns to be pinned programmatically (#11680)
mui/mui-x: docs/data/data-grid/column-pinning/column-pinning.md

{{"demo": "ControlPinnedColumns.js", "bg": "inline"}}

## Disable column pinning

### For all columns

Column pinning is enabled by default, but you can easily disable this feature by setting the `disableColumnPinning` prop.

```tsx
<DataGridPro disableColumnPinning />
```

### For some columns

To disable the pinning of a single column, set the `pinnable` property in `GridColDef` to `false`.

```tsx
<DataGridPro columns={[{ field: 'id', pinnable: false }]} /> // Default is `true`.
```

### Pin non-pinnable columns programmatically

> It may be desirable to allow one or more columns to be pinned (or unpinned) programmatically which don't support pinning (i.e. columns for which prop `disableColumnPinning = true` or `colDef.pinnable = false`).

Reviewer: "which don't support pinning": that's a bit confusing, I'd rather phrase it: "which can't be unpinned by the user".

> It may be desirable to allow one or more columns to be pinned (or unpinned) programmatically which don't support pinning (i.e. columns for which prop `disableColumnPinning = true` or `colDef.pinnable = false`).

Author: ```suggestion
It may be desirable to allow one or more columns to be pinned or unpinned programmatically that can not be pinned or unpinned on the UI (i.e. columns for which prop `disableColumnPinning = true` or `colDef.pinnable = false`).
```
